src/test/java/org/mitre/synthea/engine/GeneratorTest.java

```diff
@@ -251,4 +267,83 @@ public void run() {
 
     generateThread.interrupt();
   }
-}
+  
+  @Test
+  public void testUpdateAfterCreation() throws Exception {
+    // Get 100 people
+    Generator.GeneratorOptions opts = new Generator.GeneratorOptions();
+    opts.population = 1;
+    opts.minAge = 50;
+    opts.maxAge = 100;
+    Generator generator = new Generator(opts);
+    final int NUM_RECS = 10;
+    Person[] people = new Person[NUM_RECS];
+    for (int i = 0; i < NUM_RECS; i++) {
+      long personSeed = UUID.randomUUID().getMostSignificantBits() & Long.MAX_VALUE;
+      Random randomForDemographics = new Random(personSeed);
+      Map<String, Object> demoAttributes = generator.randomDemographics(randomForDemographics);
+      people[i] = generator.createPerson(personSeed, demoAttributes);
+      generator.recordPerson(people[i], i);
+    }
+    
+    // Update them for 10 years in the future
+    generator.stop = generator.stop + Utilities.convertTime("years", 10);
+    for (Person p: people) {
+      generator.updatePerson(p);
+    }
+  }
+  
+  /**
+   * Serialize an array of people, then deserialize and return them. Note that when serializing
+   * more than one person it is much more efficient to serialize them within a collection since
+   * shared objects will only be serialized once and deserialization will not create duplicate
+   * objects in memory.
+   * @param original array of people to serialize
+   * @return new array of people following serialization and deserialization process
+   */
+  public static Person[] serializeAndDeserialize(Person[] original)
+          throws IOException, ClassNotFoundException {
+    // Serialize
+    File tf = File.createTempFile("patient", "synthea");
+    FileOutputStream fos = new FileOutputStream(tf);
+    ObjectOutputStream oos = new ObjectOutputStream(fos);
+    oos.writeObject(original);
+    oos.close();
+    fos.close();
+    
+    // Deserialize
+    FileInputStream fis = new FileInputStream(tf);
+    ObjectInputStream ois = new ObjectInputStream(fis);
+    Person[] rehydrated = (Person[])ois.readObject();
+    
+    return rehydrated;
+  }
+
+
+  @Test
+  public void testUpdateAfterCreationAndSerialization() throws Exception {
```

Comment:
One test that I would like to see...
```
- 1 patient is generated.
- Patient is serialized.
- Patient record is exported (e.g. FHIR R4)
- Patient is deserialized.
- Patient is run into future.
- Patient record is exported again (e.g. same format, FHIR R4)
- Assert that portion of first exported record is identical to overlapping portion of second exported record
```

Reply:
Have done this manually. Problem is that all of the ids and references to those ids are different between the two exports.

Reply:
Something else we need to fix then. Added #682